oauth2_http/java/com/google/auth/oauth2/CertificateIdentityPoolSubjectTokenSupplier.java

```diff
@@ -97,33 +118,157 @@ private static String encodeCert(X509Certificate certificate)
 
   /**
    * Retrieves the X509 subject token. This method loads the leaf certificate specified by the
-   * {@code credentialSource.credentialLocation}. The subject token is constructed as a JSON array
-   * containing the base64-encoded (DER format) leaf certificate. This JSON array serves as the
-   * subject token for mTLS authentication.
+   * {@code credentialSource.credentialLocation}. If a trust chain path is configured in the {@code
+   * credentialSource.certificateConfig}, it also loads and includes the trust chain certificates.
+   * The subject token is constructed as a JSON array containing the base64-encoded (DER format)
+   * leaf certificate, followed by the base64-encoded (DER format) certificates in the trust chain.
+   * This JSON array serves as the subject token for mTLS authentication.
    *
    * @param context The external account supplier context. This parameter is currently not used in
    *     this implementation.
-   * @return The JSON string representation of the base64-encoded leaf certificate in a JSON array.
-   * @throws IOException If an I/O error occurs while reading the certificate file.
+   * @return The JSON string representation of the base64-encoded certificate chain (leaf
+   *     certificate followed by the trust chain, if present).
+   * @throws IOException If an I/O error occurs while reading the certificate file(s).
    */
   @Override
   public String getSubjectToken(ExternalAccountSupplierContext context) throws IOException {
+    String leafCertPath = credentialSource.getCredentialLocation();
```

Comment:
nit: No need to address in this PR, but it feels like the getCredentialLocation method should be renamed to getLeafCertPath() for clarity (unless it's a shared abstraction for non-X509 use cases as well?). The fact that the leaf cert is used as "Credentials" w.r.t to X.509 WIF is more of a higher-level business logic that's better addressed elsewhere.